unit_tests/src/test_replica_truncate.cpp

```diff
@@ -0,0 +1,962 @@
+
+            // Attempt to truncate the object.
+            REQUIRE(0 == rc_replica_truncate(&comm, &truncate_doi, &output_str));
+
+            // Ensure that the replicas in the replication resource hierarchy were updated.
+            CHECK(GOOD_REPLICA == replica::replica_status(comm, target_object, 1));
+            CHECK(new_size == replica::replica_size(comm, target_object, 1));
+            CHECK(original_mtime_replica_1 != replica::last_write_time(comm, target_object, 1));
+            CHECK(GOOD_REPLICA == replica::replica_status(comm, target_object, 2));
+            CHECK(new_size == replica::replica_size(comm, target_object, 2));
+            CHECK(original_mtime_replica_2 != replica::last_write_time(comm, target_object, 2));
+
+            // Ensure that the replica outside of the replication resource hierarchy was not updated and is now stale.
+            CHECK(STALE_REPLICA == replica::replica_status(comm, target_object, 0));
+            CHECK(contents.size() == replica::replica_size(comm, target_object, 0));
+            CHECK(original_mtime_replica_0 == replica::last_write_time(comm, target_object, 0));
+        }
+
+        SECTION("smaller_size")
+        {
+            // Target the replication resource so that the replicas therein may be targeted for truncate.
+            addKeyVal(&truncate_doi.condInput, RESC_NAME_KW, replication_resc.c_str());
+
+            constexpr auto new_size = contents.size() - 1;
+            //constexpr auto new_contents = std::string_view{contents.data(), new_size};
+            truncate_doi.dataSize = new_size;
+
+            // Attempt to truncate the object.
+            REQUIRE(0 == rc_replica_truncate(&comm, &truncate_doi, &output_str));
+
+            // Ensure that the replicas in the replication resource hierarchy were updated.
+            CHECK(GOOD_REPLICA == replica::replica_status(comm, target_object, 1));
+            CHECK(new_size == replica::replica_size(comm, target_object, 1));
+            CHECK(original_mtime_replica_1 != replica::last_write_time(comm, target_object, 1));
+            CHECK(GOOD_REPLICA == replica::replica_status(comm, target_object, 2));
+            CHECK(new_size == replica::replica_size(comm, target_object, 2));
+            CHECK(original_mtime_replica_2 != replica::last_write_time(comm, target_object, 2));
+
+            // Ensure that the replica outside of the replication resource hierarchy was not updated and is now stale.
+            CHECK(STALE_REPLICA == replica::replica_status(comm, target_object, 0));
+            CHECK(contents.size() == replica::replica_size(comm, target_object, 0));
+            CHECK(original_mtime_replica_0 == replica::last_write_time(comm, target_object, 0));
+        }
+    }
+    catch (const irods::exception& e) {
+        fmt::print(stderr, "irods::exception occurred: [{}]", e.what());
+    }
+    catch (const std::exception& e) {
+        fmt::print(stderr, "std::exception occurred: [{}]", e.what());
+    }
```

Comment:
If we keep these, consider making them fail the test if reached.

Reply:
I will remove these along with the others if that is what we decide to do.

Reply:
Removed try-catch blocks.